tests/test_cte.py

```diff
@@ -711,3 +713,132 @@ def test_django52_ambiguous_column_names(self):
             ('venus', 22, "admin"),
             ('venus', 23, "admin"),
         ])
+
+    def test_django52_queryset_aggregates_klass_error(self):
+        cte = CTE(
+            Order.objects.annotate(user_name=F("user__name"))
+            .values("user_name")
+            .annotate(c=Count("user_name"))
+            .values("user_name", "c")
+        )
+        qs = with_cte(cte, select=cte)
+        # Executing the query should not raise TypeError: 'NoneType' object is not subscriptable
+        self.assertEqual(list(qs), [{"user_name": "admin", "c": 22}])
+
+    def test_django52_annotate_model_field_name_after_queryset(self):
+        # Select the `id` field in one CTE
+        cte = CTE(Order.objects.values("id", "region", "user_id"))
+        # In the next query, when querying from the CTE we reassign the `id` field
+        # Previously, this would have thrown an exception
+        qs = (
+            with_cte(cte, select=cte)
+            .annotate(id=F('user_id'))
+            .values_list('id', 'region')
+            .order_by('id', 'region')
+            .distinct()
+        )
+        self.assertEqual(list(qs), [
+            (1, 'earth'),
+            (1, 'mars'),
+            (1, 'mercury'),
+            (1, 'moon'),
+            (1, 'proxima centauri'),
+            (1, 'proxima centauri b'),
+            (1, 'sun'),
+            (1, 'venus'),
+        ])
+
+    if django.VERSION >= (5, 2):
+        def test_queryset_after_values_list(self):
```

Comment:
What do you think of changing these to use `pytest.mark.skipif`?

```suggestion
    @pytest.mark.skipif(django.VERSION < (5, 2), reason="Requires Django 5.2+")
    def test_queryset_after_values_list(self):
```

All tests in this `if` block will need to be unindented and decorated, so this code suggestion is incomplete.